core/src/main/java/org/apache/carbondata/core/datastore/page/SafeFixLengthColumnPage.java

```diff
@@ -283,8 +283,8 @@ public BigDecimal getDecimal(int rowId) {
   @Override
   public byte[][] getByteArrayPage() {
     byte[][] data = new byte[arrayElementCount][];
-    for (int i = 0; i < arrayElementCount; i++) {
-      data[i] = fixedLengthData[i];
+    if (arrayElementCount >= 0) {
```

Comment:
```suggestion
    if (arrayElementCount = 0) {
```

Reply:
change to "if (arrayElementCount = 0) {"